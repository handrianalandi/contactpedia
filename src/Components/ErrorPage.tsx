import React from 'react'
import errorVectorImage from '../assets/error_vector.png'
import styled from '@emotion/styled'
import { Link, useNavigate } from 'react-router-dom'
import Button from './Button'

const ErrorPageWrapper = styled.div`
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    gap: 10px;
    width: 100%;
`
const ErrorImage = styled.img`
    width: 50vw;
    max-width: 250px;
`

const ErrorTitle = styled.h2`
    font-size: 1.5rem;
    font-weight: 700;
    margin: 0;
    font-style: italic;
    text-align: center;
`

const ErrorDescription = styled.p`
    font-size: 1rem;
    font-weight: 500;
    margin: 0;
    text-align: center;
`

const ErrorLink = styled(Link)`
    color: #06ba63;
`

export default function ErrorPage() {
    const navigate = useNavigate();
    const goBack = () => {
        navigate('/');
    }
    const reload = () => {
        navigate(window.location.pathname);
    }
  return (
    <ErrorPageWrapper>
        <ErrorImage src={errorVectorImage} alt="Error" />
        <ErrorTitle>Oops! Something went wrong.</ErrorTitle>
        <ErrorDescription>Please <ErrorLink to={'#'} onClick={reload}>refresh</ErrorLink> this page or try again later</ErrorDescription>
        <Button onClick={goBack}>Back to Home</Button>
    </ErrorPageWrapper>
  )
}
