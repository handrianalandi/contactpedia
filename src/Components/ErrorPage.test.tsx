import { render } from "@testing-library/react";
import ErrorPage from "./ErrorPage";
import { MemoryRouter } from "react-router-dom";
import { createSerializer } from "@emotion/jest";

expect.addSnapshotSerializer(createSerializer());

const mockedUsedNavigate = jest.fn();

jest.mock("react-router-dom", () => ({
  ...(jest.requireActual("react-router-dom") as any),
  useNavigate: () => mockedUsedNavigate,
}));

describe("ErrorPage", () => {
  it("should render correctly", () => {
    const { container } = render(
      <MemoryRouter>
        <ErrorPage />
      </MemoryRouter>
    );
    expect(container).toMatchSnapshot();
  });
});
