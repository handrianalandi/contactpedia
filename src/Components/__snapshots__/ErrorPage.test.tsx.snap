// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ErrorPage should render correctly 1`] = `
<div>
  <div
    class="css-15228ja"
  >
    <img
      alt="Error"
      class="css-30l6hb"
      src="error_vector.png"
    />
    <h2
      class="css-nyxvcy"
    >
      Oops! Something went wrong.
    </h2>
    <p
      class="css-g6iv93"
    >
      Please 
      <a
        class="css-1skq4v4"
        href="/"
      >
        refresh
      </a>
       this page or try again later
    </p>
    <button
      class="btn btn-primary"
      type="button"
    >
      Back to Home
    </button>
  </div>
</div>
`;
